pkg/controller/seed-controller-manager/default-application-controller/controller.go

```diff
@@ -334,19 +336,27 @@ func (r *Reconciler) generateApplicationInstallation(application appskubermaticv
 		annotations[appskubermaticv1.ApplicationDefaultedAnnotation] = "true"
 	}
 
+	namespace := application.Name
+	config, err := r.configGetter(ctx)
+	if err != nil {
+		r.log.Warnf("failed to check kubermatic configuration default settings for applications: %w", err)
+	}
+	if config != nil {
+		if config.Spec.UserCluster.DefaultApplications.Namespace != "" {
+			namespace = config.Spec.UserCluster.DefaultApplications.Namespace
+		}
+	}
+
+	appNamespace := r.getAppNamespace(ctx, application)
 	app := appskubermaticv1.ApplicationInstallation{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        application.Name,
-			Namespace:   application.Name,
+			Namespace:   namespace,
 			Annotations: annotations,
 			Labels:      application.Labels,
 		},
 		Spec: appskubermaticv1.ApplicationInstallationSpec{
-			Namespace: appskubermaticv1.AppNamespaceSpec{
-				Name: application.Name,
-				// This ensures that the namespace is created in the user cluster, if it doesn't already exist.
-				Create: true,
```

Comment:
If I see this correctly then getAppNamespace for a non-empty DefaultNamespace is just whatever is in that spec. This in turn means that if DefaultNamespace.Create was false in the original (which I think is default if left empty), would result in create being false and then the namespace will not get created. Or am I missing something?

(in case I am missing nothing): I wonder why the unit test is not detecting this, afaics it should throw an error when the namespace does not exist (unless fakeClientProvider somehow handles this for you somehow) 🤔

Reply:
This was moved to the new function `getAppNamespace`. In [ce](https://github.com/kubermatic/kubermatic/blob/70bdd1cca20da9488d38395e56625e6252b82643/pkg/controller/seed-controller-manager/default-application-controller/wrappers_ce.go#L27-L32) exactly the same is returned by the function. In [ee](https://github.com/kubermatic/kubermatic/blob/70bdd1cca20da9488d38395e56625e6252b82643/pkg/controller/seed-controller-manager/default-application-controller/wrappers_ee.go#L28-L30) a setting in the app definition will be used instead if it is configured.